modules/openapi-generator/src/main/resources/php-symfony/serialization/StrictJsonDeserializationVisitor.mustache

```diff
@@ -15,58 +15,145 @@
+    ) {
+        $this->jsonDeserializationVisitor = new JsonDeserializationVisitor($options, $depth);
+    }
+
     /**
      * {@inheritdoc}
      */
-    public function visitString($data, array $type, Context $context)
+    public function visitNull($data, array $type) 
+    {
+        return $this->jsonDeserializationVisitor->visitNull($data, $type);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function visitString($data, array $type): string
     {
         if (!is_string($data)) {
-            throw TypeMismatchException::fromValue('string', $data, $context);
+            throw TypeMismatchException::fromValue('string', $data);
         }
 
-        return parent::visitString($data, $type, $context);
+        return $this->jsonDeserializationVisitor->visitString($data, $type);
     }
 
     /**
      * {@inheritdoc}
      */
-    public function visitBoolean($data, array $type, Context $context)
+    public function visitBoolean($data, array $type): bool
     {
         if (!is_bool($data)) {
-            throw TypeMismatchException::fromValue('boolean', $data, $context);
+            throw TypeMismatchException::fromValue('boolean', $data);
         }
 
-        return parent::visitBoolean($data, $type, $context);
+        return $this->jsonDeserializationVisitor->visitBoolean($data, $type);
     }
 
     /**
      * {@inheritdoc}
      */
-    public function visitInteger($data, array $type, Context $context)
+    public function visitInteger($data, array $type): int
     {
         if (!is_int($data)) {
-            throw TypeMismatchException::fromValue('integer', $data, $context);
+            throw TypeMismatchException::fromValue('integer', $data);
         }
 
-        return parent::visitInteger($data, $type, $context);
+        return $this->jsonDeserializationVisitor->visitInteger($data, $type);
     }
 
     /**
      * {@inheritdoc}
      */
-    public function visitDouble($data, array $type, Context $context)
+    public function visitDouble($data, array $type): float
     {
         if (!is_float($data) && !is_integer($data)) {
-            throw TypeMismatchException::fromValue('double', $data, $context);
+            throw TypeMismatchException::fromValue('double', $data);
         }
 
-        return parent::visitDouble($data, $type, $context);
+        return $this->jsonDeserializationVisitor->visitDouble($data, $type);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function visitArray($data, array $type): array
+    {
+        return $this->jsonDeserializationVisitor->visitArray($data, $type);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function visitDiscriminatorMapProperty($data, ClassMetadata $metadata): string
+    {
+        return $this->jsonDeserializationVisitor->visitDiscriminatorMapProperty($data, $type);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function startVisitingObject(ClassMetadata $metadata, object $object, array $type): void
```

Comment:
2nd param changed from `object $data` to `object $object `compared to the base method